test/functional/p2p_sendrecon.py

```diff
@@ -0,0 +1,165 @@
+        pre_wtxid_version_msg.nServices = P2P_SERVICES
+        pre_wtxid_version_msg.relay = 1
+        peer.send_message(pre_wtxid_version_msg)
+        peer.wait_for_verack()
+        assert not peer.sendrecon_msg_received
+
+        self.log.info('SENDRECON for fRelay=false should not be sent')
+        peer = self.nodes[0].add_p2p_connection(SendReconReceiver(), send_version=False, wait_for_verack=False)
+        no_txrelay_version_msg = msg_version()
+        no_txrelay_version_msg.nVersion = P2P_VERSION
+        no_txrelay_version_msg.strSubVer = P2P_SUBVERSION
+        no_txrelay_version_msg.nServices = P2P_SERVICES
+        no_txrelay_version_msg.relay = 0
+        peer.send_message(no_txrelay_version_msg)
+        peer.wait_for_verack()
+        assert not peer.sendrecon_msg_received
+
+        self.log.info('valid SENDRECON received')
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(create_sendrecon_msg())
+        self.log.info('second SENDRECON triggers a disconnect')
+        peer.send_message(create_sendrecon_msg())
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with initiator=responder=0 triggers a disconnect')
+        sendrecon_no_role = create_sendrecon_msg()
+        sendrecon_no_role.initiator = False
+        sendrecon_no_role.responder = False
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(sendrecon_no_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with initiator=0 and responder=1 from inbound triggers a disconnect')
+        sendrecon_wrong_role = create_sendrecon_msg(initiator=False)
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(sendrecon_wrong_role)
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON with version=0 triggers a disconnect')
+        sendrecon_low_version = create_sendrecon_msg()
+        sendrecon_low_version.version = 0
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(sendrecon_low_version)
+        peer.wait_for_disconnect()
+
+        self.log.info('sending SENDRECON after sending VERACK triggers a disconnect')
+        # We use PeerNoVerack even though verack is sent right after, to make sure it was actually
+        # sent before sendrecon is sent.
+        peer = self.nodes[0].add_p2p_connection(PeerNoVerack(), send_version=True, wait_for_verack=False)
+        peer.send_message(msg_verack())
+        peer.send_message(create_sendrecon_msg())
+        peer.wait_for_disconnect()
+
+        self.log.info('SENDRECON without WTXIDRELAY is ignored (recon state is erased after VERACK)')
+        with self.nodes[0].assert_debug_log(['Forget reconciliation state of peer=8']):
```

Comment:
I'd suggest leaving out the peer id, since that isn't necessarily a correctness issue and could change if we add more tests
```suggestion
        with self.nodes[0].assert_debug_log(['Forget reconciliation state of peer']):
```